### PR TITLE
#410 - 1 Add inbound detail view api

### PR DESCRIPTION
**Pull request by joshxg**
#410 

- This is the API for the detail view
- also a few util changes

### CODE REVIEW

File: packages/invoices/src/InboundShipment/DetailView/api.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
+import {
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/DetailView/api.ts:1]: same as the outbound with casting to the inbound gql types

File: packages/invoices/src/OutboundShipment/DetailView/api.ts [hunk 2]
```diff
@@ -79,50 +70,6 @@ const stockLineGuard = (stockLine: StockLineResponse): StockLineNode => {
   throw new Error('Unknown');
 };
 
-export const getMutation = (): string => gql`
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/api.ts:82]: offroad removal - unused

File: packages/invoices/src/utils.ts [hunk 3]
```diff
@@ -1,16 +1,58 @@
 import {
   LocaleKey,
   InvoiceNodeStatus,
-  InvoiceNodeType,
   useTranslation,
+  InvoiceNodeType,
```
Comment (joshxg) [inline packages/invoices/src/utils.ts:5]: few off roads and changes for upcoming prs - probably shouldn't have put all of this into this pr

File: packages/invoices/src/InboundShipment/DetailView/api.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      const { updateInboundShipments } = batchInboundShipment;
+      if (
+        updateInboundShipments?.[0]?.__typename ===
+        'UpdateInboundShipmentResponseWithId'
+      ) {
+        return patch;
+      }
+    }
+
+    throw new Error(':shrug');
```
Comment (mark-prins) [inline packages/invoices/src/InboundShipment/DetailView/api.ts:193]: 😉

**Review by mark-prins — APPROVED** [review]
👍  Looks great!

**Pull request merged**